src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -730,7 +730,7 @@ internal ProtocolToken NextMessage(ReadOnlySpan<byte> incomingBuffer)
                     NetEventSource.Info(this, "NextMessage() returned SecurityStatusPal.CredentialsNeeded");
 
                 SetRefreshCredentialNeeded();
-                status = GenerateToken(incomingBuffer, ref nextmsg);
+                status = GenerateToken(ReadOnlySpan<byte>.Empty, ref nextmsg);
```

Comment:
This looks wrong. While the `incomingBuffer` buffer is empty on Unix it can have data via `ReplyOnReAuthenticationAsync`

Reply:
The issue was that when we passed `incomingBuffer` second time after creds refresh, it appended the data second time in to OpenSSL and later failed due to some error. I wonder why the same does not happen on windows though. I wll work around this some other way.

Reply:
Since we now invoke cert re-selection inline, I can revert this change